docs/reference/performance.rst

```diff
@@ -0,0 +1,277 @@
+resources of the GPU, from the individual multiprocessors to the device as a
+whole.
+
+.. _memory optimization:
+Memory optimization
+===============================================================================
+The first step in maximizing memory throughput is to minimize low-bandwidth 
+data transfers. This involves reducing data transfers between the host and the 
+device, as these have lower bandwidth than transfers between global memory and 
+the device.
+
+Additionally, data transfers between global memory and the device should be 
+minimized by maximizing the use of on-chip memory: shared memory and caches. 
+Shared memory acts as a user-managed cache, where the application explicitly 
+allocates and accesses it. A common programming pattern is to stage data from 
+device memory into shared memory. This involves each thread of a block loading 
+data from device memory to shared memory, synchronizing with all other threads 
+of the block, processing the data in shared memory, synchronizing again if 
+necessary, and writing the results back to device memory.
+
+For some applications, a traditional hardware-managed cache is more appropriate
+to exploit data locality. On devices of certain compute capabilities, the same 
+on-chip memory is used for both L1 and shared memory, and the amount dedicated 
+to each is configurable for each kernel call.
+
+Finally, the throughput of memory accesses by a kernel can vary significantly 
+depending on the access pattern for each type of memory. Therefore, the next 
+step in maximizing memory throughput is to organize memory accesses as 
+optimally as possible. This is especially important for global memory accesses,
+as global memory bandwidth is low compared to available on-chip bandwidths and
+arithmetic instruction throughput. Thus, non-optimal global memory accesses
+generally have a high impact on performance.
+
+Data Transfer
+-------------------------------------------------------------------------------
+applications should aim to minimize data transfers between the host and the 
+device. This can be achieved by moving more computations from the host to the 
+device, even if it means running kernels that do not fully utilize the 
+parallelism for device. Intermediate data structures can be created, used, 
+and discarded in device memory without being mapped or copied to host memory.
+
+Batching small transfers into a single large transfer can improve performance 
+due to the overhead associated with each transfer. On systems with a front-side
+bus, using page-locked host memory can enhance data transfer performance.
+
+When using mapped page-locked memory, there is no need to allocate device 
+memory or explicitly copy data between device and host memory. Data transfers 
+occur implicitly each time the kernel accesses the mapped memory. For optimal
+performance, these memory accesses should be coalesced, similar to global
+memory accesses.
```

Comment:
There is no definition of "coalescing" here or before, but it is mentioned in such a way like it was introduced earlier. It should be accompanied with explanation or at least linked to explanation in another place.

Reply:
Introduction is added.